### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by vm1000**
@evanagsaccenture 
Index check is created.
Not sure about 3. task - if there should be printed "employee is out of office" even if that employee was at office in the beginning or like I did - it deletes all employees that match employee and then prints "employee is out of office".

### CODE REVIEW

File: assingment.playground/Contents.swift [hunk 1]
```diff
@@ -2,5 +2,41 @@ import Foundation
...
+}
+
+public func workIsLeftBy(employee: String) {
+    for index in 0 ... workplace.seats.count - 1 {
+        let seat = workplace.seats[index]
+        if seat == employee {
+            workplace.seats[index] = nil
+        }
+    }
+        print("\(employee) seems to be out-of-office.")
```
Comment (ghost) [inline assingment.playground/Contents.swift:40]: This is not correct. Remove this line.

File: assingment.playground/Contents.swift [hunk 2]
```diff
@@ -2,5 +2,41 @@ import Foundation
...
+            print("\(employee) tried to take seat number \(index + 1), unfortunatelly that seat was occupied by \(oldEmployee).")
+        }
+    }
+    } else {
+            print ("Something went wrong! index must be between 0 and \(workplace.seats.count - 1)!")
+    }
+}
+
+public func workIsLeftBy(employee: String) {
+    for index in 0 ... workplace.seats.count - 1 {
```
Comment (ghost) [inline assingment.playground/Contents.swift]: Let's re-read the criteria:

> Write a method named `workIsLeftBy` that takes in a parameter named `employee` of type `String`. Method should iterate through currently present workers, and all employees that match `employee` should leave workplace making seats available. If there are no employees that match `employee` output a message **"`employee` seems to be out-of-office."**.

Therefore, we understand that before we start iterating through array of seats, we need to check [if there is a seat occupied by such `employee`](https://stackoverflow.com/questions/24102024/how-to-check-if-an-element-is-in-an-array). If there is we do the remaining logic - iterate and and ask employees to leave, otherwise, we print the error message.

File: assingment.playground/Contents.swift [hunk 3]
```diff
@@ -2,5 +2,41 @@ import Foundation
...
+    let seat = workplace.seats[seatIndex]
+    if seat == nil {
+        return true
+    } else {
+        return false
+    }
+}
+
+public func occupyEmptySeat(index: Int, employee: String = "👩🏻‍💼") {
+    if index >= 0 && index < workplace.seats.count {
```
Comment (ghost) [inline assingment.playground/Contents.swift:18]: **Fun fact:** this line can be written shorter `if 0 ..< workplace.seats.count ~= index {`

_It validated the variable `index` is in range between 0 and `workplace.seats.count - 1`_
_Vai mainīgais `index` pieder kopai [0; x), kur x - `workplace` objekta, `seats` mainīgā elementu skaits_

File: assingment.playground/Contents.swift [hunk 4]
```diff
@@ -2,5 +2,41 @@ import Foundation
 
 private let workplace = Workplace()
 
-workplace.setEmptySeat(count: <#Int#>)
-workplace.employeesInOffice(employeeCount: <#Int#>)
+workplace.setEmptySeat(count: 9)
+workplace.employeesInOffice(employeeCount: 6)
+
+public func isSeatAvailable(_ seatIndex: Int) -> Bool {
+    let seat = workplace.seats[seatIndex]
```
Comment (ghost) [inline assingment.playground/Contents.swift:9]: **Fun fact:** This could have been written as one-liner
`return workplace.seats[seatIndex] == nil ? true : false`

It is convenience way to write `if-else` where blocks execute only one command.
Formula goes as follows `is condition true` ? `yes` : `no`.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Fun facts are simply for your information, do not act on those

File: assingment.playground/Contents.swift [hunk 5]
```diff
@@ -29,13 +29,16 @@ public func occupyEmptySeat(index: Int, employee: String = "👩🏻‍💼") {
 }
 
 public func workIsLeftBy(employee: String) {
+    if workplace.seats.contains(employee) {
     for index in 0 ... workplace.seats.count - 1 {
```
Comment (ghost) [inline assingment.playground/Contents.swift]: Please fix indentation, you can let Xcode take care of that, select given method and use shortcut `control` + `i`

File: assingment.playground/Contents.swift [hunk 6]
```diff
@@ -3,7 +3,7 @@ import Foundation
 private let workplace = Workplace()
 
 workplace.setEmptySeat(count: 9)
-workplace.employeesInOffice(employeeCount: 6)
+workplace.employeesInOffice(employeeCount: 2)
```
Comment (ghost) [inline assingment.playground/Contents.swift:6]: Why was this changed?
Avoid unnecessary changes.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by vm1000** [comment]
Thanks for review! 
I fixed ndentation.
Made those changes with employee count just to check if method works properly.

**Review by ghost — APPROVED** [review]
🚀

**Comment by ghost** [comment]
@vm1000 now that pull request has been approved you can go ahead and marge pull request. Confirm merge with descriptive comment and delete branch. I did this step this time.
Afterwards, in _Sourcetree_, you checkout master branch, you'll see that file is in state it was before the solution, you pull in new changes and voila!
